### PR TITLE
Validate custom license length.

### PR DESCRIPTION
**Pull request by JaxkDev**
Fairly self explanatory, majority of plugins submitted have no license by taking advantage of custom license option and leaving it blank.
(This is left blank when poggit finds no LICENSE file as well)

PR adds a length check if license is custom, it check against < 200char (200 because the smallest OSL license i could find was the Fair license which *without user details* measures in at about 206).

### CODE REVIEW

File: src/poggit/release/submit/PluginSubmission.php [hunk 1]
```diff
@@ -267,6 +267,10 @@ private function strictValidate() {
         }
         if($this->license->type !== "custom") {
             // TODO validate license name from GitHub
+        } else {
+            if(strlen($this->license->custom) < 200){
+                throw new SubmitException("Custom license's must be under an open source license, and the full text of the license is required.");
```
Comment (SOF3) [inline src/poggit/release/submit/PluginSubmission.php]: what's up with the apostrophe here?

**Review by SOF3 — CHANGES_REQUESTED** [review]
How about validation at client side?

**Comment by JaxkDev** [comment]
Ok

**Comment by JaxkDev** [comment]
Tested custom license of < 200 chars, above 200 and non-custom licenses.

image

**Comment by SOF3** [comment]
Is closer compiler doing something fishy? The diff seemed to contain something irrelevant.

File: js/newSubmit.min.js [hunk 2]
```diff
@@ -10,21 +10,21 @@ this.$getRow().find(".form-input-react").text()}},!1,"submit"!==f.mode));g.push(
...
+g.push("Plugin association");g.push(new k("assocParent",f.fields.assocParent,"Associate Release of:","submit2-assoc-parent",N,function(){return{valid:!0}}));0<f.assocChildren.length&&g.push(new k("assocChildrenUpdates",f.fields.assocChildren,"Children Associate Releases","submit2-assoc-children-updates",H(f.assocChildren,!0),function(){return{valid:!0}}));g.push("Other details");g.push(new k("license",f.fields.license,"License","submit2-license",O,function(){return null!==this.getValue().custom&&
+200>this.getValue().custom.length?{valid:!1,message:"The custom license must be a open source license, and the full text is required."}:{valid:!0}}));g.push(new k("perms",f.fields.perms,"Permissions","submit2-perms",H(f.consts.perms,!0),function(){return{valid:!0}}));g.push(E=new k("authors",f.fields.authors,"Producers","submit2-authors",P,function(){return{valid:this.getValue().length>=E.type.minRows,message:"There must be at least 1 producer!"}}));t(F);setTimeout(q,500,E);var b=$(".form-table");
```
Comment (JaxkDev) [inline js/newSubmit.min.js:14]: When removing my code (back to original js only this is removed rest of changes persist.

**Pull request merged**